sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetEncodingSuite.scala

```diff
@@ -78,4 +78,30 @@ class ParquetEncodingSuite extends ParquetCompatibilityTest with SharedSQLContex
       }}
     }
   }
+
+  test("Read row group containing both dictionary and plain encoded pages") {
+    spark.conf.set("parquet.dictionary.page.size", "2048")
+    spark.conf.set("parquet.page.size", "4096")
+
+    withTempPath { dir =>
+      // In order to explicitly test for SPARK-14217, we set the parquet dictionary and page size
+      // such that the following data spans across 3 pages (within a single row group) where the
+      // first page is dictionary encoded and the remaining two are plain encoded.
+      val data = (0 until 512).flatMap(i => Seq.fill(3)(i.toString))
+      data.toDF("f").coalesce(1).write.parquet(dir.getCanonicalPath)
+      val file =
+        SpecificParquetRecordReaderBase.listDirectory(dir).toArray.head.asInstanceOf[String]
+
+      val reader = new VectorizedParquetRecordReader
+      reader.initialize(file, null /* set columns to null to project all columns */)
```

Comment:
`VectorizedParquetRecordReader` is a Java class instead of a Scala class, so named parameter isn't feasible here.